rs/ic_os/config/src/lib.rs

```diff
@@ -31,8 +44,380 @@ fn parse_config_line(line: &str) -> Option<(String, String)> {
 pub fn config_map_from_path(config_file_path: &Path) -> Result<ConfigMap> {
     let file_contents = read_to_string(config_file_path)
         .with_context(|| format!("Error reading file: {}", config_file_path.display()))?;
-    Ok(file_contents
+
+    let normalized_file_contents = file_contents.replace("\r\n", "\n").replace("\r", "\n");
+
+    Ok(normalized_file_contents
         .lines()
         .filter_map(parse_config_line)
         .collect())
 }
+
+// Prefix should have a max length of 19 ("1234:6789:1234:6789")
+// It could have fewer characters though. Parsing as an ip address with trailing '::' should work.
+fn is_valid_ipv6_prefix(ipv6_prefix: &str) -> bool {
+    ipv6_prefix.len() <= 19 && format!("{ipv6_prefix}::").parse::<Ipv6Addr>().is_ok()
+}
+
+pub fn get_config_ini_settings(config_file_path: &Path) -> Result<(NetworkSettings, bool)> {
```

Comment:
Could you add a type for this, too? It's only two fields but a struct with named fields is still easier to read/use/extend than a tuple.

Reply:
Added!